x-pack/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/json_diff/json_diff.test.tsx

```diff
@@ -0,0 +1,290 @@
+ * @returns {boolean} `true` if the element contains the specified text and none of its children do,
+ * otherwise `false`.
+ * Utilizes Lodash's `escapeRegExp` to handle special characters in the `strings` array.
+ */
+const matchInOrder =
+  (strings: string[]): Matcher =>
+  (content: string, element: Element | null) => {
+    if (element) {
+      const hasText = (node: Element): boolean => {
+        const regex = new RegExp(strings.map((string) => escapeRegExp(string)).join('.*?'));
+        return regex.test(node.textContent || '');
+      };
+
+      const nodeHasText = hasText(element);
+      const childrenDontHaveText = Array.from(element.children).every((child) => !hasText(child));
+
+      return nodeHasText && childrenDontHaveText;
+    }
+
+    return false;
+  };
+
+describe('Rule upgrade workflow: viewing rule changes in JSON diff view', () => {
+  it('User can see precisely how property values would change after upgrade', () => {
+    const removedBackgroundColor = 'rgb(255, 235, 233)';
+    const removedAccentColor = 'rgba(255, 129, 130, 0.4)';
+    const addedBackgroundColor = 'rgb(230, 255, 236)';
+    const addedAccentColor = 'rgb(171, 242, 188)';
+
+    const oldRule: RuleResponse = {
+      ...savedRuleMock,
+    };
+
+    const newRule: RuleResponse = {
+      ...savedRuleMock,
+    };
+
+    /* Changes to test line update */
+    oldRule.version = 1;
+    newRule.version = 2;
+
+    /* Changes to test line removal */
+    oldRule.author = ['Alice', 'Bob', 'Charlie'];
+    newRule.author = ['Alice', 'Charlie'];
+
+    /* Changes to test line addition */
+    delete oldRule.license;
+    newRule.license = 'GPLv3';
+
+    const { getByText } = render(<RuleDiffTab oldRule={oldRule} newRule={newRule} />);
+
+    /* LINE UPDATE */
+    const updatedLine = getByText(matchInOrder(['-', 'version', '1', '+', 'version', '2']));
+
+    const updatedLineBefore = within(updatedLine).getByText(matchInOrder(['version', '1']));
+    expect(updatedLineBefore).toHaveStyle(`background: ${removedBackgroundColor}`);
+
+    const updatedWordBefore = within(updatedLineBefore).getByText('1');
+    expect(updatedWordBefore).toHaveStyle(`background: ${removedAccentColor}`);
+
+    const updatedLineAfter = within(updatedLine).getByText(matchInOrder(['version', '2']));
+    expect(updatedLineAfter).toHaveStyle(`background: ${addedBackgroundColor}`);
+
+    const updatedWordAfter = within(updatedLineAfter).getByText('2');
+    expect(updatedWordAfter).toHaveStyle(`background: ${addedAccentColor}`);
+
+    /* LINE REMOVAL */
+    /* Check that the removed line is displayed only once */
+    expect(screen.queryAllByText(matchInOrder(['Bob']))).toHaveLength(1);
+
+    /* Removed line appears in the "before" column */
+    const removedLine = getByText(matchInOrder(['-', 'Bob']));
+
+    const removedLineBefore = within(removedLine).getByText(matchInOrder(['Bob']));
+    expect(removedLineBefore).toHaveStyle(`background: ${removedBackgroundColor}`);
+
```

Comment:
nit: do you think it's possible to capture also `removedLineAfter` and show that it has no color?
So as to completely cover these cases:
```
Examples:
| change_type | column         | bg_color         | accent_color         | line_badge |
| removed     | Current rule   | removed_bg_color | none                 | -          |
| removed     | Elastic update | none             | none                 | none       |
```

Reply:
Addressed in a follow-up commit